crMeta/crMetadata.go

```diff
@@ -618,10 +642,10 @@ func getMetaFromMCResponse(lookupResp *base.SubdocLookupResponse) (cas uint64, c
 }
 
 // This will find the custom CR XATTR from the req body
-func getMetaFromMCRequest(req *mc.MCRequest) (cas uint64, cvSrc hlv.DocumentSourceId, cvVer uint64, pvMap, mvMap hlv.VersionsMap, err error) {
+func getMetaFromMCRequest(req *mc.MCRequest) (cas, cvCas uint64, cvSrc hlv.DocumentSourceId, cvVer uint64, pvMap, mvMap hlv.VersionsMap, err error) {
 	cas = binary.BigEndian.Uint64(req.Extras[16:24])
 	if req.DataType&mcc.XattrDataType == 0 {
-		return cas, "", 0, nil, nil, nil
+		return cas, 0, "", 0, nil, nil, nil
 	}
 	body := req.Body
```

Comment:
Code below is duplicate code with duplicate areas for improvements.
Can we look into simplifying and avoid code reuse? 

Something like:
`func getMetaFromBody`
Both MCReq and MCResp just have to grab the body and call the same function and pass it in to get all the meta info.

Reply:
Moved common code to parseHlvFields().